client/state/current-user/selectors.js

```diff
@@ -44,6 +44,21 @@ export function getCurrentUserLocale( state ) {
 }
 
 /**
+ * Returns the number of sites for the current user.
+ *
+ * @param  {Object}  state  Global state tree
+ * @return {?Number}        Current user site count
+ */
+export function getCurrentUserSiteCount( state ) {
+	const user = getCurrentUser( state );
+	if ( ! user ) {
+		return null;
+	}
+
+	return user.site_count || null;
```

Comment:
If we have a valid user but they actually don't have any sites (user only account), wouldn't `0` make more sense here?